portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,29 +14,83 @@
 
 package com.google.sps.servlets;
 
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
-import com.google.gson.Gson;
+// import com.google.sps.data.Task;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  ArrayList<String> Comments = new ArrayList<String>();
   @Override
+  // Writes in Comment Arraylist to /data
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/json;");
-   // response.getWriter().println("Hello Samuel!");
+    response.setContentType("data/json");
 
-    ArrayList<String> HardCodedMessagesForJson = new ArrayList<String>();
-    HardCodedMessagesForJson.add("Hmm does this work?");
-    HardCodedMessagesForJson.add("I guess this works?");
-    HardCodedMessagesForJson.add("It does work!");
+    Query query = new Query("Comment").addSort("Timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> new_entries = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("Comment");
+      long timestamp = (long) entity.getProperty("Timestamp");
+
+      String new_comment = comment + " : " + timestamp + "\n";
+      new_entries.add(new_comment);
+    }
 
     Gson gson = new Gson();
-    String json = gson.toJson(HardCodedMessagesForJson);
+    String json = gson.toJson(new_entries);
     response.getWriter().println(json);
   }
+  // Takes in Post request from index.html and adds content to Comment ArrayList
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Where is IOException thrown out from your code?

Reply:
It's kinda weird. When I try to remove that part of the function header then I get the compilation error: "unreported exception java.io.IOException; must be caught or declared to be thrown". While reading through different StackOverflow forums about it I found this one: https://stackoverflow.com/questions/11589302/why-is-throws-exception-necessary-when-calling-a-function . 
Apparently Java requires that if you don't manually handle an exception using a try catch then you must put the "throws exception" in the function header. I think its just a precautionary measure.

Reply:
Sorry to confuse you. I didn't mean deleting this IOException from the function signature. My question was where this IOException throwing out from the code.

The reason why a function needs to throw/cath an exception is because inside the function, somewhere in the code throws it. For example, IOException is usually related to open/close a file. If you have a function like this

public void foo () throws IOException {
 ...
   bar();
...
}

private void bar() throws IOException {
 // open a file
... // do sth
// close a file
}

The reason foo() throws/cath IOException is because it calls bar() inside and bar() throws IOException, but if foo() doesn't call bar(), then in its code, IOException is never thrown out. It doesn't need to IOException at all. 

Throwing an Exception on a function where in the code it's never thrown out will also make some compilers fail as well.